docs/Reference-for-template.json.md

```diff
@@ -52,31 +52,275 @@ You can find a sample of a complex `template.json` file here [Sample](https://gi
+
+##### Examples
+Boolean optional parameter with default value `false`:
+```json
+  "symbols": {
+    "TestProject": {
+      "type": "parameter",
+      "dataType": "bool",
+      "defaultValue": "false"
+    }
+  },
+```
+ 
+Choice optional parameter with 3 possible values:
+```json
+  "symbols": {
+    "Framework": {
+      "type": "parameter",
+      "description": "The target framework for the project.",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "netcoreapp3.1",
+          "description": "Target netcoreapp3.1"
+        },
+        {
+          "choice": "netstandard2.1",
+          "description": "Target netstandard2.1"
+        },
+        {
+          "choice": "netstandard2.0",
+          "description": "Target netstandard2.0"
+        }
+      ],
+      "replaces": "netstandard2.0",
+      "defaultValue": "netstandard2.0"
+    }
+}
+```
+ 
+String optional parameter, replaces TargetFrameworkOverride:
+```json
+ "symbols": {
+    "TargetFrameworkOverride": {
+      "type": "parameter",
+      "description": "Overrides the target framework",
+      "replaces": "TargetFrameworkOverride",
+      "datatype": "string",
+      "defaultValue": ""
+    }
+}
+```
+
+#### Derived symbol
+
+A symbol that defines transformation of another symbol.  The value of this symbol is derived from the value of another symbol by the application of form defined in `valueTransform`.
+
+|Name|Description|
+|---|---|
+|`type`|`derived`|
+|`valueSource`|The name of the other symbol whose value will be used to derive this value.|
+|`valueTransform`|The name of the value form to apply to the source value.|
+|`replaces`|The text to replace with the value of this symbol.|	 
+|`fileRename`|The fileRename element defines the filenames which will be replaced by symbol value.| 
+|`description`|Human readable text describing the meaning of the symbol. This has no effect on template generation.|
+
+##### Examples
+Renames `Application1` file to value of `name` symbol after last dot
+```json
+{
+...
+  "symbols": {
+    "app1Rename": {
+      "type": "derived",  
+      "valueSource": "name",  
+      "valueTransform": "ValueAfterLastDot",  
+      "fileRename": "Application1",  
+      "description": "A value derived from the 'name' param, used to rename Application1.cs"
+    }
+  }
+...
+  "forms": {
+    "ValueAfterLastDot": {
+      "identifier": "replace",
+      "pattern": "^.*\\.(?=[^\\.]+$)",        // regex to match everything up to and including the final "."
+      "replacement": ""  // replace it with empty string
+    }
+  }
+...
+}
+```
+
+#### Generated symbol
```

Comment:
By code, derived symbol can also have the following fields: `binding`, `onlyIf`. Does they make any sense for generated symbols?